### ISSUE TITLE
<type 'exceptions.ValueError'>: list.remove(x): x not in list

### ISSUE THREAD

**Issue by Dieterbe** [segmentio/analytics-python#69]
Hi,
we signed up for your service and are trying it out for the first time.
i wrote a simple script (not threaded) that sends some data but it crashes with this error, which seems to be caused by your library.

```
Exception in thread Thread-1 (most likely raised during interpreter shutdown):
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
  File "/usr/local/lib/python2.7/dist-packages/analytics/consumer.py", line 28, in run
  File "/usr/local/lib/python2.7/dist-packages/analytics/consumer.py", line 39, in upload
  File "/usr/local/lib/python2.7/dist-packages/analytics/consumer.py", line 62, in next
  File "/usr/local/lib/python2.7/dist-packages/analytics/consumer.py", line 78, in next_item
  File "/usr/lib/python2.7/Queue.py", line 177, in get
  File "/usr/lib/python2.7/threading.py", line 363, in wait
<type 'exceptions.ValueError'>: list.remove(x): x not in list
```

**Comment by f2prateek** [comment]
Can you share the script? So we can reproduce the failing case.

**Comment by Dieterbe** [comment]
here you go https://gist.github.com/Dieterbe/5895835cdee93b807ad8

**Comment by jgershen** [comment]
@Dieterbe Thanks for reporting this error! It is fixed in version 1.2.4 of this library.

A quick (additional) note -- when writing scripts like your sample gist, you'll want to also call `flush` before your script exits: https://segment.com/docs/libraries/python/#how-do-i-flush-right-now-
This will make sure any pending calls have been uploaded to Segment's servers before exiting.
